app/src/main/java/protect/card_locker/LoyaltyCard.java

```diff
@@ -97,7 +156,103 @@ public void writeToParcel(Parcel parcel, int i) {
+            balanceType = tmpBalanceType != null ? Currency.getInstance(tmpBalanceType) : null;
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_CARD_ID)) {
+            cardId = bundle.getString(BUNDLE_LOYALTY_CARD_CARD_ID);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_BARCODE_ID)) {
+            barcodeId = bundle.getString(BUNDLE_LOYALTY_CARD_BARCODE_ID);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_BARCODE_TYPE)) {
+            String tmpBarcodeType = bundle.getString(BUNDLE_LOYALTY_CARD_BARCODE_TYPE, null);
+            barcodeType = tmpBarcodeType != null ? CatimaBarcode.fromName(tmpBarcodeType) : null;
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_HEADER_COLOR)) {
+            int tmpHeaderColor = bundle.getInt(BUNDLE_LOYALTY_CARD_HEADER_COLOR, -1);
+            headerColor = tmpHeaderColor != -1 ? tmpHeaderColor : null;
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_STAR_STATUS)) {
+            starStatus = bundle.getInt(BUNDLE_LOYALTY_CARD_STAR_STATUS);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS)) {
+            archiveStatus = bundle.getInt(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_LAST_USED)) {
+            lastUsed = bundle.getLong(BUNDLE_LOYALTY_CARD_LAST_USED);
+        }
+        if (bundle.containsKey(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL)) {
+            zoomLevel = bundle.getInt(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL);
+        }
+    }
+
+    public Bundle toBundle() {
+        Bundle bundle = new Bundle();
+
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ID, id);
+        bundle.putString(BUNDLE_LOYALTY_CARD_STORE, store);
+        bundle.putString(BUNDLE_LOYALTY_CARD_NOTE, note);
+        if (validFrom != null) {
+            bundle.putLong(BUNDLE_LOYALTY_CARD_VALID_FROM, validFrom.getTime());
+        }
+        if (expiry != null) {
+            bundle.putLong(BUNDLE_LOYALTY_CARD_EXPIRY, expiry.getTime());
+        }
+        bundle.putString(BUNDLE_LOYALTY_CARD_BALANCE, balance.toString());
+        if (balanceType != null) {
+            bundle.putString(BUNDLE_LOYALTY_CARD_BALANCE_TYPE, balanceType.toString());
+        }
+        bundle.putString(BUNDLE_LOYALTY_CARD_CARD_ID, cardId);
+        bundle.putString(BUNDLE_LOYALTY_CARD_BARCODE_ID, barcodeId);
+        if (barcodeType != null) {
+            bundle.putString(BUNDLE_LOYALTY_CARD_BARCODE_TYPE, barcodeType.name());
+        }
+        if (headerColor != null) {
+            bundle.putInt(BUNDLE_LOYALTY_CARD_HEADER_COLOR, headerColor);
+        }
+        bundle.putInt(BUNDLE_LOYALTY_CARD_STAR_STATUS, starStatus);
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ARCHIVE_STATUS, archiveStatus);
+        bundle.putLong(BUNDLE_LOYALTY_CARD_LAST_USED, lastUsed);
+        bundle.putInt(BUNDLE_LOYALTY_CARD_ZOOM_LEVEL, zoomLevel);
+
+        return bundle;
+    }
+
+    public static LoyaltyCard fromCursor(Cursor cursor) {
```

Comment:
FYI: it's always harder for me to follow the code when the fields declared in the class, parameters for the constructor, and fields set in various methods don't all use the same ordering.

Reply:
I believe the ordering should be consistent now, nice catch, thanks. If it's still inconsistent somewhere please point out exactly where because it's kinda hard to read this code sometimes